server/src/main/java/org/elasticsearch/health/HealthIndicatorResult.java

```diff
@@ -21,22 +26,35 @@ public record HealthIndicatorResult(
     HealthIndicatorDetails details,
     List<HealthIndicatorImpact> impacts,
     List<Diagnosis> diagnosisList
-) implements ToXContentObject {
-
+) implements ChunkedToXContent {
     @Override
-    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
-        builder.startObject();
-        builder.field("status", status.xContentValue());
-        builder.field("symptom", symptom);
-        if (details != null && HealthIndicatorDetails.EMPTY.equals(details) == false) {
-            builder.field("details", details, params);
-        }
-        if (impacts != null && impacts.isEmpty() == false) {
-            builder.field("impacts", impacts);
-        }
+    public Iterator<? extends ToXContent> toXContentChunked() {
+        Iterator<? extends ToXContent> diagnosisIterator = Collections.emptyIterator();
         if (diagnosisList != null && diagnosisList.isEmpty() == false) {
-            builder.field("diagnosis", diagnosisList);
+            diagnosisIterator = diagnosisList.stream()
+                .flatMap(s -> StreamSupport.stream(Spliterators.spliteratorUnknownSize(s.toXContentChunked(), Spliterator.ORDERED), false))
+                .iterator();
```

Comment:
Would it be possible to use the 
```java
Iterators.concat(
  diagnosisList.stream().map(ChunkedToXContent::toXContentChunked).toArray(Iterator[]::new)
);
```
pattern here instead of using Spliterators? I wonder if this concat method would be more readable than the spliterator route? I assume there's no performance concern since the same pattern was used in another chunked object already.